ui/src/Header/Header.tsx

```diff
@@ -86,19 +86,39 @@ function Header({ hideSpaceButtons, hideAllButtons }: HeaderProps): JSX.Element
     return (
```

Comment:
This header would benefit from some refactoring into smaller components (within the same file is fine), but that shouldn't block this PR.